File: algorithms/0242-valid-anagram/solution.py
# Valid Anagram — count letters with a 26-slot list; lengths must match first.
# Time O(n), Space O(1).

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        counts = [0] * 26
        for i in range(len(s)):
            counts[ord(s[i]) - ord("a")] += 1
            counts[ord(t[i]) - ord("a")] -= 1
        return all(count == 0 for count in counts)

File: algorithms/0242-valid-anagram/Solution.kt
// Valid Anagram — count letters with IntArray(26); lengths must match first.
// Time O(n), Space O(1).

class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        if (s.length != t.length) return false
        val counts = IntArray(26)
        for (i in s.indices) {
            counts[s[i] - 'a']++
            counts[t[i] - 'a']--
        }
        return counts.all { it == 0 }
    }
}
